satpy/readers/agri_l1.py

```diff
@@ -0,0 +1,206 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Advanced Geostationary Radiation Imager for the Level_1 HDF format
+
+The files read by this reader are described in the official Real Time Data Service:
+
+    http://fy4.nsmc.org.cn/data/en/data/realtime.html
+
+"""
+
+import logging
+import numpy as np
+import xarray as xr
+import dask.array as da
+from datetime import datetime
+from pyresample import geometry
+from satpy.readers.hdf5_utils import HDF5FileHandler
+
+logger = logging.getLogger(__name__)
+
+# info of 500 m, 1 km, 2 km and 4 km data
+_resolution_list = [500, 1000, 2000, 4000]
+_COFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_CFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+_LOFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_LFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+
+
+class HDF_AGRI_L1(HDF5FileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDF_AGRI_L1, self).__init__(filename, filename_info, filetype_info)
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Load a dataset."""
+        logger.debug('Reading in get_dataset %s.', dataset_id.name)
+        file_key = ds_info.get('file_key', dataset_id.name)
+        lut_key = ds_info.get('lut_key', dataset_id.name)
+        data = self.get(file_key)
+        lut = self.get(lut_key)
+        if data.ndim >= 2:
+            data = data.rename({data.dims[-2]: 'y', data.dims[-1]: 'x'})
+
+        # convert bytes to string
+        data.attrs['long_name'] = data.attrs['long_name'].decode('gbk')
+        data.attrs['band_names'] = data.attrs['band_names'].decode('gbk')
+        if ds_info['file_type'] != 'agri_l1_4000m_geo':
+            data.attrs['center_wavelength'] = data.attrs['center_wavelength'].decode('gbk')
+
+        # calibration
+        calibration = ds_info['calibration']
+
+        if calibration == 'counts':
+            data.attrs['units'] = ds_info['units']
+            ds_info['valid_range'] = data.attrs['valid_range']
+            return data
+
+        elif calibration == 'reflectance':
+            logger.debug("Calibrating to reflectances")
+            # using the corresponding SCALE and OFFSET
+            cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
+            num_channel = self.get(cal_coef).shape[0]
+            if num_channel == 1:
+                # only channel_2, resolution = 500 m
+                slope = self.get(cal_coef)[:, 0].values

```

Comment:
Do these need to have the `.values`? I suppose in the long run it isn't a big deal but keeping things lazy is something I usually try for.

Reply:
@djhoese `.values` returns just one number. If I keep it lazy, I need to expand the 1D array to 2D array which has the same shape as data. Maybe, is it better to use `.values`?